compute/cloud-client/src/main/java/compute/snapshotschedule/EditSnapshotSchedule.java

```diff
@@ -0,0 +1,114 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute.snapshotschedule;
+
+// [START compute_snapshot_schedule_edit]
+import com.google.cloud.compute.v1.Operation;
+import com.google.cloud.compute.v1.Operation.Status;
+import com.google.cloud.compute.v1.PatchResourcePolicyRequest;
+import com.google.cloud.compute.v1.ResourcePoliciesClient;
+import com.google.cloud.compute.v1.ResourcePolicy;
+import com.google.cloud.compute.v1.ResourcePolicySnapshotSchedulePolicyRetentionPolicy;
+import com.google.cloud.compute.v1.ResourcePolicySnapshotSchedulePolicySchedule;
+import com.google.cloud.compute.v1.ResourcePolicySnapshotSchedulePolicySnapshotProperties;
+import com.google.cloud.compute.v1.ResourcePolicyWeeklyCycle;
+import com.google.cloud.compute.v1.ResourcePolicyWeeklyCycleDayOfWeek;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class EditSnapshotSchedule {
+
+  public static void main(String[] args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "YOUR_PROJECT_ID";
+    // Name of the region where your snapshot schedule is located.
+    String region = "us-central1";
+    // Name of the snapshot schedule you want to update.
+    String snapshotScheduleName = "YOUR_SCHEDULE_NAME";
+
+    editSnapshotSchedule(projectId, region, snapshotScheduleName);
+  }
+
+  public static Status editSnapshotSchedule(
+          String projectId, String region, String snapshotScheduleName)
+          throws IOException, InterruptedException, ExecutionException, TimeoutException {
+    String description = "Updated description11";
+    Map<String, String> snapshotLabels = new HashMap<>();
+    snapshotLabels.put("key", "value");
+    ResourcePolicyWeeklyCycleDayOfWeek dayOfWeek = ResourcePolicyWeeklyCycleDayOfWeek.newBuilder()
+            .setDay("Tuesday")
+            .setStartTime("09:00")
+            .build();
+    ResourcePolicyWeeklyCycle weeklySchedule = ResourcePolicyWeeklyCycle.newBuilder()
+            .addDayOfWeeks(dayOfWeek)
+            .build();
+    String onSourceDiskDelete = "apply-retention-policy";
+    int maxRetentionDays = 3;
+
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    try (ResourcePoliciesClient resourcePoliciesClient = ResourcePoliciesClient.create()) {
+      ResourcePolicy existingSchedule = resourcePoliciesClient
+              .get(projectId, region, snapshotScheduleName);
```

Comment:
Do we need to get the existing schedule to set the new properties? The REST example in the docs looks like they override the old properties.

Have you tried constructing the patch request directly by setting the new properties, instead of retrieving the old ones and clearing them?